libs/site-base/src/components/MainPageTransition/index.tsx

```diff
@@ -1,12 +1,26 @@
 import * as React from 'react';
-import { ContentHeightMeasure } from '../ContentHeightMeasure';
+import { useLocation } from 'react-router';
 import { ContentFader } from '../ContentFader';
 import { ContentHeightAnimate } from '../ContentHeightAnimate';
+import { ContentHeightMeasure } from '../ContentHeightMeasure';
+import { RouteMap, AuthSwitch } from '@thecointech/shared/containers/AuthRoute';
+import styles from './styles.module.less';
 
-export const MainPageTransition: React.FC = (props) => (
-  <ContentHeightAnimate>
-    <ContentFader>
-      <ContentHeightMeasure>{props.children}</ContentHeightMeasure>
-    </ContentFader>
-  </ContentHeightAnimate>
-);
+type Props = {
+  routes: RouteMap,
+  contentId?: string,
+};
+export const MainPageTransition: React.FC<Props> = ({ contentId, routes }) => {
+  const location = useLocation()
+  return (
+    <ContentHeightAnimate>
+      <ContentFader location={location}>
+        <ContentHeightMeasure>
+          <section id={contentId} className={styles.pageMainInner}>
```

Comment:
I don't like that this shared class now injects the main page section in here, but it was necessary to make sure the location is propagated correctly (and it reduces duplicate code, so not all bad)